### PR TITLE
chore(Messaggi a valore legale): [IAMVL-8] Add types, actions, store, reducers & saga entrypoint

### PR DESCRIPTION
**Pull request by fabriziofff**
## Short description
This pr adds the types, actions, store, reducers and saga entrypoint required for the legal messages.

## List of changes proposed in this pull request
- Added `ts/types/digitalInformationUnit.ts` in order to have unambiguous types to measure the digital information
- Added `ts/types/contentType.ts` in order to have a unique enumeration in the app for content-types 
- Defined the datamodel, based on the required data. This could be change in future based also on the API specification. 
`MvlData` represents the MVL with all the information required to render it. It is composed by a body (`MvlBody`), a list of possible attachments (`ReadonlyArray<MvlAttachment>`) and some metadata (`MvlMetadata`). All the `MvlData` have an unique `MvlId`, defined as unique tagged, in order to avoid to mix generic `string` with MVL ids.
- Created a new async action `mvlDetailsLoad`, in order to represent the request and the outcome of the remote loading.
- Create a new store section `byId` under `features/mvl`, defined as `IndexedById<pot.Pot<MvlData, Error>>` and used to index all the requests and results of the MVL loading.
Schermata 2021-12-02 alle 12 45 35

- Added the related reducer & selector
- Added `watchMvlSaga` as entrypoint for all the (maybe future) saga watchers related to MVL
- Added `handleGetMvl` saga stub, that will handle the `mvlDetailsLoad.request`, will generate a remote request and will transform the remote data in `MvlData`
- Added the configuration to `ts/store/reducers/backoffError.ts` in order to support the back-off.
- Added a new exception to `scripts/check_urls/check_urls.py` in order to have false positive with the mock file `ts/features/mvl/types/__mock__/mvlMock.ts`

## How to test
**Jest**
- run `ts/features/mvl/store/reducers/__test__/byId.test.ts`

**Manual**
- Enable `MVL_ENABLED`
- Dispatch a `mvlDetailsLoad.request`
- Observe the store changes

https://user-images.githubusercontent.com/26501317/144417697-a790ac24-c45c-49c0-8170-9135214b3b68.mov

### CODE REVIEW

File: ts/features/mvl/store/reducers/__test__/byId.test.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import { mvlFromIdSelector } from "../byId";
+
+const mockFailure: WithMVLId<NetworkError> = {
+  id: mvlMockId,
+  ...getGenericError(new Error("A generic error"))
+};
+
+const errorFromFailure = new Error(getNetworkErrorMessage(mockFailure));
+
+describe("byAuthCode reducer & selector behaviour", () => {
```
Comment (Undermaken) [inline ts/features/mvl/store/reducers/__test__/byId.test.ts:22]: Maybe it is a copy/paste error

File: ts/features/mvl/types/mvlData.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+};
+
+/**
+ * Additional metadata
+ * TODO: Just an initial stub, should be completed and refined
+ */
+export type MvlMetadata = {
+  sender: string;
+  receiver: string;
+  cc: ReadonlyArray<string>;
```
Comment (Undermaken) [inline ts/features/mvl/types/mvlData.ts:44]:
```suggestion
  sender: string;
  receiver: string;
  cc: ReadonlyArray<string>;
```
Could it be an improvement using the existing type `EmailAddress`?

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/features/mvl/store/reducers/__test__/byId.test.ts (hunk 1 above)
Comment (fabriziofff) [reply ts/features/mvl/store/reducers/__test__/byId.test.ts]: Yes 😅, changed in b723ca4f04630e9d89e0957f28e60efd0bf88620

File: ts/features/mvl/types/mvlData.ts (hunk 2 above)
Comment (fabriziofff) [reply ts/features/mvl/types/mvlData.ts]: Better for sure, the more types the merrier!
Changed in a8093521bcab81608fea29e2d159eb8087aa5f17

**Pull request merged**